### PR TITLE
refactor: add missing types

### PR DESCRIPTION
**Pull request by Puckoland**

### CODE REVIEW

File: generals/game.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import numpy as np
 import gymnasium as gym
-from typing import Dict, List
+from typing_extensions import TypeAlias
+
 from generals.config import DIRECTIONS, PASSABLE, MOUNTAIN, INCREMENT_RATE
 
 from scipy.ndimage import maximum_filter
 
+Observation: TypeAlias = dict[str, np.ndarray]
```
Comment (strakam) [inline generals/game.py:10]: This is not correct (probably was during your commit) -> now Observation is [a bit more complex](https://github.com/strakam/Generals-RL/blob/661858051d6fa39de801be85b412d8eed5d9a4a6/generals/game.py#L330), but idk how to 'type' it :(

File: generals/game.py [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+Info: TypeAlias = dict[str, int]
```
Comment (strakam) [inline generals/game.py:11]: probably `dict[str, any]` is better, now `int` is just a coincidence

File: generals/integrations/pettingzoo_integration.py [hunk 3]
```diff
@@ -88,7 +92,7 @@ def step(self, action):
 
         return observations, rewards, terminated, truncated, infos
 
-    def default_rewards(self, observations):
+    def default_rewards(self, observations: dict[str, Observation]) -> dict[str, int]:
```
Comment (strakam) [inline generals/integrations/pettingzoo_integration.py:95]: reward should be `float` (`int` is a coincidence now)

**Review by strakam — CHANGES_REQUESTED** [review]
nice changes, thanks :+1: 
.. just a few comments

File: generals/game.py (hunk 1 above)
Comment (Puckoland) [reply generals/game.py]: You are right, it was changed after my commit.

File: generals/game.py (hunk 2 above)
Comment (Puckoland) [reply generals/game.py]: done

File: generals/integrations/pettingzoo_integration.py (hunk 3 above)
Comment (Puckoland) [reply generals/integrations/pettingzoo_integration.py]: done

**Review by jdujava — COMMENTED** [review]
Maybe also modify `Tuple` -> `tuple` in `README.md`.

**Comment by strakam** [comment]
will modiy `README.md` after merge
.. @Puckoland thanks!

**Pull request merged**